### ISSUE TITLE
[FR] Activate guide lines, by default, when activating modules that use it (and find a way to explain change for new users

### ISSUE THREAD

**Issue by Nilvus** [darktable-org/darktable#9490]
**Is your feature request related to a problem? Please describe.**
Previously if we have guide lines set in a module, they just show up when activate it. So just one click to activate a module and then show guide lines.

Now when we activate a module, if lines toggle button is not on, they just don't show up. So to show them, it's one click more. Just a step back and not so intuitive. For some modules (crop, crop & rotate, perspective...), having guide lines is most of the time wanted. By the way, when 3.8 will be released if we stay as it is actually, I'm quite sure we will have a lot of complaints about that new behavior.

I'm ok with the idea to have a general way to set guide lines and his new feature is great. But we need to go a step forward.

**Describe the solution you'd like**

When activate modules that use/need guide lines (I've list them above ; of course we will need to the ashift rework by @AlicVB to that list), just toggle on guide lines.

**Alternatives**

If some don't want that, just add an option in guide lines toggle menu to activate or not those when activating related modules (by default set in "on" setting of course).

**Comment by piratenpanda** [comment]
> Now when we activate a module, if lines toggle button is not on, they just don't show up. So to show them, it's one click more. Just a step back and not so intuitive.

If I activate the "show guides" option in the crop and rotate module menu it shows guides even without the toggle at the bottom activated. Or are you talking about something else?

**Comment by elstoc** [comment]
Yes it's in the "presets" menu. I'm not sure what else you're looking for here.

**Comment by Nilvus** [comment]
I didn't see that. And it's off by default. It should be on by default.

This is better with it of course but remains one thing: this change is not intuitive and will be hard for lot of users to see/understand. We need to think so about how deal with that and avoid complaints/questions about that change.

**Comment by elstoc** [comment]
Personally I would prefer it in the main module UI (rather than a menu), as I mentioned in the original PR. Current C&R behaviour is that the guides default to "None" (i.e. off). Since the "None" option has been removed from the combobox, having the checkbox default to "off" seems the best default behaviour.

**Comment by AlicVB** [comment]
**default value** : I prefer to set it to "don't show' : 
- previously default value was 'off'
- we have now more modules that can accept "auto-apply" guides. Basically all distorting ones. But apart for some of them (crop, new perspective, ...) auto-applying guides for the others is a corner (but still valid) case : architecture work, ... So if we go that way (default to 'show') we'll have to do that for some and not for others....
- I find guides distracting most of the time, especially when judging cropping or rotation overall effect, so I prefer to only show them on demand (but maybe I'm the only one here)

**where to put the "automatically show guides" checkbox ?**
putting them in global preferences is not really feasible due to the number of possibility : it will take way too much space
I've also tried to put that in the bottom toolbar menu at first, but it was messy and very difficult to understand for the user
so basically we have just 2 choices : 
1. in the hamburger menu
2. in the module UI

I prefer solution 1 because imho setting that is a "one-time" thing, and the main UI is not the best place to expose such settings (which is not part of module parameters) : it just clutter the UI and add some distractions...

At some point I've think about adding also a line in module UI, with the possibility to hide it in global preference (default = show for discoverability). That would have pleased everyone, but that has a drawback : that would have added that line in module which need it only for special usage : retouch, liquify, watermark, etc... and the will have the exact opposite complains, I fear...

Again, that's just my opinion, and maybe (certainly) I don't understand your workflow...

**Comment by elstoc** [comment]
The main use for me is changing the guides while cropping (because the guides I want to use will ultimately depend on the image and I quite often switch between rule of thirds and the various golden mean guides) and occasionally I want to switch them off or mirror them. So it's all a bit cumbersome for me at the moment. I can do it with shortcuts but ultimately I think shortcuts should be something you can "also do" rather than the recommended way to interact with the UI. So my preference is still for option 2 with a right-click popup to change the type of guide (plus making it as clear as possible that the guide settings are global).

I honestly don't see the issue with permanently adding a line to the main UI of distort modules even if the use case is rare - they're all pretty compact right now. It's only a major issue for people who like having multiple modules at once (since only liquify - the smallest - can even be used with blending). Those people are always going to have issues with vertical space.

**Comment by AlicVB** [comment]
This is not really a "space" problem, but more a "distracting" one. imho the less useless thing we expose in the UI the better ! you can then focus on really usefull ones... not to speak that with less widgets, I find it easier to "directly" find the right one, without having to search it...

That said, I see your point. I'll try to implement what I've said below so we can try it !

**Comment by elstoc** [comment]
Absolutely agree that it will be rare to use this for the non-crop modules (I won't use it, except maybe to judge rotation), but for crop I expect a lot of people consider changing guide overlays a key part of the workflow.

**Comment by AlicVB** [comment]
@Nilvus @elstoc : I've done in PR #9495 a proposal with both solutions : a combobox in the ui (that can be hidden in prefs) and the menu entry.
If you have time to test and comment, that would be great ! thanks !
